lib/core.ps1

```diff
@@ -600,29 +603,33 @@ function Invoke-ExternalCommand {
     if ($RunAs) {
         $Process.StartInfo.UseShellExecute = $true
         $Process.StartInfo.Verb = 'RunAs'
-    } else {
-        $Process.StartInfo.CreateNoWindow = $true
-    }
-    if ($FilePath -match '^((cmd|cscript|wscript|msiexec)(\.exe)?|.*\.(bat|cmd|js|vbs|wsf))$') {
-        $Process.StartInfo.Arguments = $ArgumentList -join ' '
-    } elseif ($Process.StartInfo.ArgumentList.Add) {
-        # ArgumentList is supported in PowerShell 6.1 and later (built on .NET Core 2.1+)
-        # ref-1: https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist?view=net-6.0
-        # ref-2: https://docs.microsoft.com/en-us/powershell/scripting/whats-new/differences-from-windows-powershell?view=powershell-7.2#net-framework-vs-net-core
-        $ArgumentList | ForEach-Object { $Process.StartInfo.ArgumentList.Add($_) }
-    } else {
-        # escape arguments manually in lower versions, refer to https://docs.microsoft.com/en-us/previous-versions/17w5ykft(v=vs.85)
-        $escapedArgs = $ArgumentList | ForEach-Object {
-            # escape N consecutive backslash(es), which are followed by a double quote, to 2N consecutive ones
-            $s = $_ -replace '(\\+)"', '$1$1"'
-            # escape N consecutive backslash(es), which are at the end of the string, to 2N consecutive ones
-            $s = $s -replace '(\\+)$', '$1$1'
-            # escape double quotes
-            $s = $s -replace '"', '\"'
-            # quote the argument
-            "`"$s`""
+    }
+    if ($Silent) {
+        $Process.StartInfo.UseShellExecute = $true
+        $Process.StartInfo.WindowStyle = [System.Diagnostics.ProcessWindowStyle]::Hidden
+    }
+    if ($ArgumentList.Length -gt 0) {
```

Comment:
Does `if ($ArgumentList)` work?

Reply:
Not always. When `$ArgumentList` contains only one empty string, we expect the result of `if ($ArgumentList)` to be `$true`, while it is actually `$false`. Just like both `$false -eq @('')` and `$false -eq @()` will output `True`.